html5ever/src/tree_builder/mod.rs

```diff
@@ -1172,17 +1172,17 @@ impl<Handle, Sink> TreeBuilder<Handle, Sink>
 
         declare_tag_set!(listed = [form_associatable] - "img");
 
-        // Step 7.
-        let qname = QualName::new(None, ns, name);
-        let elem = create_element(&mut self.sink, qname.clone(), attrs.clone());
-
         let insertion_point = self.appropriate_place_for_insertion(None);
         let (node1, node2) = match insertion_point {
             LastChild(ref p) |
             BeforeSibling(ref p) => (p.clone(), None),
             TableFosterParenting { ref element, ref prev_element } => (element.clone(), Some(prev_element.clone())),
         };
 
+        // Step 7.
+        let qname = QualName::new(None, ns, name);
+        let elem = create_element(&mut self.sink, qname.clone(), attrs.clone(), Some(&node1));
```

Comment:
This might cause problems, because if the `TableFosterParenting` arm is matched in the match block, then either `node1` or `node2` can be the parent. This decision is taken on Servo's side, because we need to access the DOM to figure out which node is the parent.

Reply:
Hmm true... Would elements be in the same document? Should I send both to `create_element`?

Reply:
These elements are from the [stack of open elements](https://html.spec.whatwg.org/multipage/parsing.html#stack-of-open-elements), where it is possible that two elements may not have the same document associated with them. So you need to pass both elements.

Reply:
@cbrewster Does this comment still need to be addressed?

Reply:
I believe so, its been a while since I worked on these changes. I'm a bit busy at the moment with school, but I'll get back to this in a couple weeks.